docs/source/general/new_task.rst

```diff
@@ -0,0 +1,177 @@
+******************************
+Adding a new task from scratch
+******************************
+
+One should start by subclassing :class:`~flash.core.model.Task`.
+
+To make this new task work for training, one should have to override the forward function.
+
+.. code-block:: python
+
+    from flash.core.model import Task
+
+    class ImageClassifier(Task):
+
+        def __init__(
+            self,
+            num_classes,
+            backbone="resnet18",
+            loss_fn: Callable = F.cross_entropy,
+        ):
+            super().__init__(
+                model=None,
+                loss_fn=loss_fn,
+            )
+
+            self.save_hyperparameters()
+
+            self.backbone, num_features = instantiate_backbone(backbone)
```

Comment:
where this `instantiate_backbone` come from? is Flash function?
probably user function, then add a comment...